steps/elasticsearch/session.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Telefonica/golium"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/tidwall/sjson"
+)
+
+// Session contains the information of a elasticsearch session.
+type Session struct {
+	Client       *elasticsearch.Client
+	SearchResult golium.Map
+	Correlator   string
+}
+
+// ConfigureClient creates a elasticsearch connection based on the URI.
+func (s *Session) ConfigureConnection(ctx context.Context, config elasticsearch.Config) error {
+	var err error
+	if s.Client, err = elasticsearch.NewClient(config); err != nil {
+		return errors.Wrap(err, "failed configuring elasticsearch client")
+	}
+	s.Correlator = uuid.NewString()
+	return nil
+}
+
+// CreatesDocument creates a document in index with given JSON properties.
```

Comment:
CreateDocument  creates a document, with the given JSON and for the given index.

Reply:
Is not better `NewDocument`?